smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ClientConfigGenerator.kt

```diff
@@ -140,6 +140,7 @@ class ClientConfigGenerator(
                         prop.propertyType.default
                     )
                 }
+                is ClientConfigPropertyType.Custom -> prop.propertyType.render(prop, ctx.writer)
```

Comment:
**Style**: Since we're already switching on `prop.propertyType`, this and other repeated references to `prop.propertyType` could be cleaned up with a temporary capture:

```kotlin
when (val type = prop.propertyType) {
    is ClientConfigPropertyType.Custom -> type.render(prop, ctx.writer)
}
```